src/features/task/taskSlice.js

```diff
@@ -18,8 +18,19 @@ const taskSlice = createSlice({
 
             state.tasks.splice(deletableTaskIndex, 1);
         },
+        markAsDone: (state, action) => {
+            const taskIdToMarkAsDone = action.payload;
+            const taskIndexToMarkAsDone = state.tasks.findIndex((task) => task.id === taskIdToMarkAsDone);
+
+            if (taskIndexToMarkAsDone === -1) {
+                return;
+            }
+
+            state.tasks[taskIndexToMarkAsDone].isCompleted = true;
+            state.tasks[taskIndexToMarkAsDone].completedAt = new Date().toISOString();
```

Comment:
avoid mutating the Redux state. Use a shallow copy to update the value. Thank you!

// https://redux.js.org/faq/immutable-data#why-is-immutability-required-by-redux

Reply:
I'm using Redux-Toolkit here, it uses [Immer](https://immerjs.github.io/immer/) inside it to manage state immutably. So, here the parameter `state` is actually a **draft** of the actual `state`. We are changing the draft state, and `immer` is immutably changing the actual state. 
Here is the [reference](https://redux-toolkit.js.org/api/createReducer#direct-state-mutation:~:text=To%20make%20things%20easier%2C%20createReducer%20uses%20immer%20to%20let%20you%20write%20reducers%20as%20if%20they%20were%20mutating%20the%20state%20directly.%20In%20reality%2C%20the%20reducer%20receives%20a%20proxy%20state%20that%20translates%20all%20mutations%20into%20equivalent%20copy%20operations.). Thank you!